diff --git a/quantum/encoder.h b/quantum/encoder.h
index 7f692bc78..c9519e9d4 100644
--- a/quantum/encoder.h
+++ b/quantum/encoder.h
@@ -23,6 +23,7 @@ void encoder_init(void);
 bool encoder_read(void);

 bool encoder_update_kb(uint8_t index, bool clockwise);
+bool encoder_preprocess_kb(uint8_t index, bool clockwise);
 bool encoder_update_user(uint8_t index, bool clockwise);
 void encoder_inerrupt_read(uint8_t index);

diff --git a/quantum/encoder.c b/quantum/encoder.c
index 3358fec4f..7d7621f5a 100644
--- a/quantum/encoder.c
+++ b/quantum/encoder.c
@@ -80,6 +80,10 @@ __attribute__((weak)) bool encoder_update_kb(uint8_t index, bool clockwise) {
     return encoder_update_user(index, clockwise);
 }

+__attribute__((weak)) bool encoder_preprocess_kb(uint8_t index, bool clockwise) {
+    return true;
+}
+
 __attribute__((weak)) bool should_process_encoder(void) {
     return is_keyboard_master();
 }
@@ -188,7 +192,8 @@ static bool encoder_update(uint8_t index, uint8_t state) {
             if (should_process_encoder())
 #endif // SPLIT_KEYBOARD
 #ifdef ENCODER_MAP_ENABLE
-                encoder_exec_mapping(index, ENCODER_COUNTER_CLOCKWISE);
+                if (encoder_preprocess_kb(index, ENCODER_COUNTER_CLOCKWISE))
+                    encoder_exec_mapping(index, ENCODER_COUNTER_CLOCKWISE);
 #else  // ENCODER_MAP_ENABLE
         encoder_update_kb(index, ENCODER_COUNTER_CLOCKWISE);
 #endif // ENCODER_MAP_ENABLE
@@ -205,7 +210,8 @@ static bool encoder_update(uint8_t index, uint8_t state) {
             if (should_process_encoder())
 #endif // SPLIT_KEYBOARD
 #ifdef ENCODER_MAP_ENABLE
-                encoder_exec_mapping(index, ENCODER_CLOCKWISE);
+                if (encoder_preprocess_kb(index, ENCODER_CLOCKWISE))
+                    encoder_exec_mapping(index, ENCODER_CLOCKWISE);
 #else  // ENCODER_MAP_ENABLE
         encoder_update_kb(index, ENCODER_CLOCKWISE);
 #endif // ENCODER_MAP_ENABLE
@@ -256,7 +262,8 @@ void encoder_update_raw(uint8_t *slave_state) {
             encoder_value[index]++;
             changed = true;
 #    ifdef ENCODER_MAP_ENABLE
-            encoder_exec_mapping(index, ENCODER_COUNTER_CLOCKWISE);
+            if (encoder_preprocess_kb(index, ENCODER_COUNTER_CLOCKWISE))
+                encoder_exec_mapping(index, ENCODER_COUNTER_CLOCKWISE);
 #    else  // ENCODER_MAP_ENABLE
             encoder_update_kb(index, ENCODER_COUNTER_CLOCKWISE);
 #    endif // ENCODER_MAP_ENABLE
@@ -266,7 +273,8 @@ void encoder_update_raw(uint8_t *slave_state) {
             encoder_value[index]--;
             changed = true;
 #    ifdef ENCODER_MAP_ENABLE
-            encoder_exec_mapping(index, ENCODER_CLOCKWISE);
+            if (encoder_preprocess_kb(index, ENCODER_CLOCKWISE))
+                encoder_exec_mapping(index, ENCODER_CLOCKWISE);
 #    else  // ENCODER_MAP_ENABLE
             encoder_update_kb(index, ENCODER_CLOCKWISE);
 #    endif // ENCODER_MAP_ENABLE
